### DISCUSSION TITLE
[QST] Indexed spatial features

### DISCUSSION THREAD

**Discussion by Norticaranda** [rapidsai/cuspatial#971] [General]
Hi
I found that @zhangjianting is in the development process of:

[WIP] quadtree indexing for point data and point-in-polygon based indexed spatial join

Do you expect finish this feature for rapidsai:branch-0.13 release only with 2D Spacial indexing or for 3D Spacial indexing (point cloud)?

If isn't possible for this version, what is the order do you have for the rest of features?: (KD Tree, Grid, R-Tree, Voronoi, etc) and especially for 3D spatial scenarios

We have intention to help if our skills allow us.

Thanks

**Comment by zhangjianting** [comment]
@Norticaranda  I just tested the indexing/join modules on NYC taxi trip data. See the test code at https://github.com/zhangjianting/cuspatial/blob/fea-quadtree-indexing/cpp/tests/join/spatial_join_nyctaxi_test.cu. The results have been partially verified by maual inspection. 
The indexing/join implementations will be sent to code review and hopefully it can catch 0.13 release date. You are welcome to test the code we have so far and report issues to make 0.13 release robust.  
Grid and R-Tree are scheduled for the end of May. KD-Tree for 2D data is similar to Quadtree.
